### PR TITLE
fix: corrected misleading log message in save_pretrained function

### PR DESCRIPTION
**Pull request by mturetskii**
# What does this PR do?

This PR extends the fix implemented in a previous PR ([#28181](https://github.com/huggingface/transformers/pull/28181)), covering all cases where the saved file name might differ from the expected `WEIGHTS_NAME`. The earlier fix did not account for scenarios where the saved file could be named `ADAPTER_WEIGHTS_NAME` or `ADAPTER_SAFE_WEIGHTS_NAME`, leaving a potential for misleading log messages. This update ensures that all such cases are covered, and the log message accurately reflects the name of the file being saved in the `save_pretrained` function.

Fixes # https://github.com/huggingface/transformers/issues/28076
## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -2490,8 +2490,7 @@ def save_pretrained(
                 save_function(shard, os.path.join(save_directory, shard_file))
 
         if index is None:
-            weights_file_name = SAFE_WEIGHTS_NAME if safe_serialization else WEIGHTS_NAME
-            path_to_weights = os.path.join(save_directory, _add_variant(weights_file_name, variant))
+            path_to_weights = os.path.join(save_directory, _add_variant(weights_name, variant))
```
Comment (amyeroberts) [inline src/transformers/modeling_utils.py:2493]: Following from L2457 - L2459, do we still want use `_add_variant` when weights_name is an adapter name like `ADAPTER_SAFE_WEIGHTS_NAME`?

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding this fix! 

Just a quick Q, otherwise LGTM!

File: src/transformers/modeling_utils.py (hunk 1 above)
Comment (mturetskii) [reply src/transformers/modeling_utils.py:2493]: Good point. I think we can definitely omit _add_variant here. If we want to use `_add_variant` with `ADAPTER_SAFE_WEIGHTS_NAME`, we should place it above these lines and, ideally, in a separate task.

**Review by amyeroberts — APPROVED** [review]
Thanks for iterating and contributing this fix!

**Pull request merged**